### PR TITLE
Fix metrics documentation

### LINKED ISSUE #720: Unify naming of domains
## 📚 Documentation

In both the "Module metrics" and "Functional metrics" subsections we are inconsistent with how we have the domains:
They should either be just the domain name or have the "metrics" postfix. IMO it should just be the domain name.

Screenshot 2022-01-06 at 11-19-19 Module metrics — PyTorch-Metrics 0 7 0dev documentation

### PR DESCRIPTION
**Pull request by AndresAlgaba**
## What does this PR do?

Fixes #720 

## Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?
- [x] Did you write any new necessary tests?

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: docs/source/references/functional.rst [hunk 1]
```diff
@@ -335,9 +335,9 @@ tweedie_deviance_score [func]
     :noindex:
 
 
-****************
-Pairwise Metrics
-****************
+********
+Pairwise
```
Comment (Borda) [inline docs/source/references/functional.rst:339]: I would keep here metrics

File: docs/source/references/modules.rst [hunk 2]
```diff
@@ -52,12 +52,12 @@ SumMetric
...
-Audio Metrics
-*************
+*****
+Audio
+*****
 
-About Audio Metrics
-~~~~~~~~~~~~~~~~~~~
+About Audio
+~~~~~~~~~~~
```
Comment (Borda) [inline docs/source/references/modules.rst:60]: I think this is not needed as it not with other section neither

File: docs/source/references/functional.rst (hunk 1 above)
Comment (AndresAlgaba) [reply docs/source/references/functional.rst]: Removed it in my latest commit.

File: docs/source/references/modules.rst (hunk 2 above)
Comment (AndresAlgaba) [reply docs/source/references/modules.rst]: Removed it in my latest commit.

File: docs/source/references/functional.rst (hunk 1 above)
Comment (AndresAlgaba) [reply docs/source/references/functional.rst]: Added it again in my latest commit.

**Review by Borda — APPROVED** [review]

**Review by SkafteNicki — APPROVED** [review]

**Pull request merged**